lib/slack/real_time/stores/store.rb

```diff
@@ -108,20 +111,29 @@ def initialize(attrs)
         # @see https://github.com/slack-ruby/slack-api-ref/blob/master/events/channel_joined.json
         on :channel_joined do |data|
           channel_id = data.channel.id
-          channel = channels[channel_id]
-          if channel
-            channel.merge!(data.channel)
+          if data.channel.is_private?
+            channel = private_channels[channel_id]
+            if channel
+              channel.merge!(data.channel)
+            else
+              private_channels[channel_id] = Models::Channel.new(data.channel)
+            end
           else
-            channels[channel_id] = Models::Channel.new(data.channel)
+            channel = public_channels[channel_id]
+            if channel
+              channel.merge!(data.channel)
+            else
+              public_channels[channel_id] = Models::Channel.new(data.channel)
+            end
```

Comment:
This logic would be a lot simpler and future proof if we just stored `channels[]`. The implementation for `private_channels` would be `channels.select(&:is_private)`. The additional notifications such as renames and unarchives are for all channels AFAIK as well. I'll merge as is, but consider this change before we release?